packages/rest-crud/src/repository-builder.ts

```diff
@@ -1,55 +0,0 @@
-// Copyright IBM Corp. 2019. All Rights Reserved.
-// Node module: @loopback/rest-crud
-// This file is licensed under the MIT License.
-// License text available at https://opensource.org/licenses/MIT
-
-import {
-  DefaultCrudRepository,
-  Entity,
-  EntityCrudRepository,
-  juggler,
-} from '@loopback/repository';
-import assert from 'assert';
-
-/**
- * Create (define) a repository class for the given model.
- *
- * @example
- *
- * ```ts
- * const ProductRepository = defineCrudRepositoryClass(Product);
- * ```
- *
- * @param modelCtor A model class, e.g. `Product`.
- */
-export function defineCrudRepositoryClass<
-  T extends Entity,
-  IdType,
-  Relations extends object = {}
->(
-  entityClass: typeof Entity & {prototype: T},
-): RepositoryClass<T, IdType, Relations> {
-  const repoName = entityClass.name + 'Repository';
-  const defineNamedRepo = new Function(
-    'EntityCtor',
-    'BaseRepository',
-    `return class ${repoName} extends BaseRepository {
-      constructor(dataSource) {
-        super(EntityCtor, dataSource);
-      }
-    };`,
-  );
-
-  // TODO(bajtos) make DefaultCrudRepository configurable (?)
```

Comment:
The idea is to allow the caller of `defineCrudRepositoryClass` to provide the base class to inherit from. For example, if they want to use their own class `AuditedEntityCrudRepository` as the repository class to be used by all models.

Reply:
Yes.